python/cusignal/radartools/radartools.py

```diff
@@ -121,3 +122,88 @@ def pulse_doppler(x, window=None, nfft=None):
         pd_dataMatrix = cp.fft.fft(x, nfft, axis=0)
 
     return pd_dataMatrix
+
+
+_new_ynorm_kernel = cp.ElementwiseKernel(
+    "int32 xlen, raw T xnorm, raw T ynorm",
+    "T out",
+    """
+    int row = i / xlen;
+    int col = i % xlen;
+    int x_col = col - ( xlen - 1 ) + row;
+
+    if ( ( x_col >= 0 ) && ( x_col < xlen ) ) {
+        out = ynorm[col] * thrust::conj( xnorm[x_col] );
+    } else {
+        out = T(0,0);
+    }
+    """,
+    "_new_ynorm_kernel",
+    options=("-std=c++11",),
+)
+
+
+def ambgfun(x, fs, prf, y=None, cut='2d', cutValue=0):
+    """
+    Calculates the normalized ambiguity function for the vector x
+
+    Parameters
+    ----------
+    x : ndarray
+        Input pulse waveform
+
+    fs: int, float
+        Sampling rate in Hz
+
+    prf: int, float
+        Pulse repetition frequency in Hz
+
+    y : ndarray
+        Second input pulse waveform. If not given, y = x
+
+    cut : string
+        Direction of one-dimensional cut through ambiguity function
+
+    cutValue : int, float
+        Time delay or doppler shift at which one-dimensional cut
+        through ambiguity function is taken
+
+    Returns
+    -------
+    amfun : ndarray
+        Normalized magnitude of the ambiguity function
+    """
+    if 'float64' in x.dtype.name:
+        x = cp.asarray(x, dtype=cp.complex128)
+    elif 'float32' in x.dtype.name:
+        x = cp.asarray(x, dtype=cp.complex64)
+    else:
+        x = cp.asarray(x)
+
+    xnorm = x / cp.linalg.norm(x)
+    if y is None:
+        y = x
+        ynorm = xnorm
+    else:
+        ynorm = y / cp.linalg.norm(y)
+
+    len_seq = len(xnorm) + len(ynorm)
+    nfreq = 2**ceil(log2(abs(len_seq - 1)))
```

Comment:
Where did this math come from? I believe the `abs` ambiguous

Reply:
You're right @mnicely. The abs was a holdover from MATLAB's nextpow2 function but is unnecessary here because the tau length is always positive.